src/compiler/factory/nodeFactory.ts

```diff
@@ -4836,8 +4834,9 @@ export function createNodeFactory(flags: NodeFactoryFlags, baseFactory: BaseNode
     }
 
     // @api
-    function createImportAttributes(elements: readonly ImportAttribute[], multiLine?: boolean): ImportAttributes {
+    function createImportAttributes(token: ImportAttributes["token"], elements: readonly ImportAttribute[], multiLine?: boolean): ImportAttributes {

```

Comment:
Why does this need to take a `token`? The `createAssertClause` function has a hard-coded token kind; can this one do the same but with the `WithToken`?

Reply:
The token was hard-coded in the `createAssertClause` for backward compatibility. The `parser` relies on `createImportAttributes`, where we need to support `assert` and `with` and emit the appropriate tokens.

Reply:
I see. I think it would be a bit unfortunate to have this API once import assertions are fully removed. Can the parser use `createImportAssertion` instead, or maybe have `token`  be the last parameter with a default?

Reply:
Is it ok to use `@internal` overload?

Reply:
I think that would be fine, personally

Reply:
Yeah, because the only reason to pass `token` to `createImportAttributes` is to make an assert clause, which API consumers can already do with `createAssertClause` which will appropriately tell them it’s deprecated.